airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py

```diff
@@ -326,6 +328,23 @@ def _render_nested_template_fields(
             self._do_render_template_fields(content, ("limits", "requests"), context, jinja_env, seen_oids)
             return
 
+        if id(content) not in seen_oids and isinstance(content, k8s.V1Volume):
+            seen_oids.add(id(content))
+            self._do_render_template_fields(
+                content, ("name", "persistent_volume_claim"), context, jinja_env, seen_oids
+            )
+            return
+
+        if id(content) not in seen_oids and isinstance(content, k8s.V1VolumeMount):
+            seen_oids.add(id(content))
+            self._do_render_template_fields(content, ("name",), context, jinja_env, seen_oids)
+            return
+
+        if id(content) not in seen_oids and isinstance(content, k8s.V1PersistentVolumeClaimVolumeSource):
+            seen_oids.add(id(content))
+            self._do_render_template_fields(content, ("claim_name",), context, jinja_env, seen_oids)
+            return
+
```

Comment:
ok so.... at this point probably best to simplify / condense a bit

here's what i came up with .


you could add your additions to this 

```python
        if id(content) not in seen_oids:
            template_fields = None
            if isinstance(content, k8s.V1EnvVar):
                template_fields = ("value", "name")
            elif isinstance(content, k8s.V1ResourceRequirements):
                template_fields = ("limits", "requests")
            if template_fields:
                seen_oids.add(id(content))
                self._do_render_template_fields(content, template_fields, context, jinja_env, seen_oids)
                return
```

Reply:
i think it keeps same logic but ....  little cleaner